<launch>
  <!-- R2, SE2, R3, or SE3 are supported -->
  <arg name="sampling_space" default="R2" />

  <node pkg="differentiable_rmap" type="NodeRmapPlanning" name="rmap_planning"
        output="screen">
    <rosparam subst_value="true">
      sampling_space: $(arg sampling_space)
      config_path: $(find differentiable_rmap)/config/RmapPlanning.yaml
      svm_path: /tmp/rmap_svm_model_$(arg sampling_space).libsvm
    </rosparam>

    <rosparam subst_value="true">
      bag_path: /tmp/rmap_grid_set_$(arg sampling_space).bag
    </rosparam>
  </node>

  <node pkg="optmotiongen_core" type="InteractiveMarkerManager.py" name="rviz_server"
        clear_params="true">
    <rosparam if="$(eval arg('sampling_space') == 'R2')">
      interactive_markers:
        - name: target
          axis_controls: [tx, ty]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [1.0, 1.0, 0.0, 0.5]
    </rosparam>

    <rosparam if="$(eval arg('sampling_space') == 'SE2')">
      interactive_markers:
        - name: target
          axis_controls: [tx, ty, rz]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [1.0, 1.0, 0.0, 0.5]
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [1.0, 1.0, 0.0, 0.5]
    </rosparam>

    <rosparam if="$(eval arg('sampling_space') == 'R3')">
      interactive_markers:
        - name: target
          axis_controls: [tx, ty, tz]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [1.0, 1.0, 0.0, 0.5]
    </rosparam>

    <rosparam if="$(eval arg('sampling_space') == 'SE3')">
      interactive_markers:
        - name: target
          axis_controls: [tx, ty, tz, rx, ry, rz]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [1.0, 1.0, 0.0, 0.5]
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [1.0, 0.0, 0.0, 0.5]
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [0.0, 1.0, 0.0, 0.5]
              pose:
                rot: [0.0, 0.0, 1.0, 1.57]
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [0.0, 0.0, 1.0, 0.5]
              pose:
                rot: [0.0, 1.0, 0.0, -1.57]
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="$(anon rviz)"
        args="-d $(find differentiable_rmap)/rviz/RmapPlanning.rviz"
        output="log" clear_params="true" />
</launch>
